CHANGELOG.md

```diff
@@ -33,6 +33,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Grant access to testclusters dir for tests ([#19085](https://github.com/opensearch-project/OpenSearch/issues/19085))
 - Fix assertion error when collapsing search results with concurrent segment search enabled ([#19053](https://github.com/opensearch-project/OpenSearch/pull/19053))
 - Fix skip_unavailable setting changing to default during node drop issue ([#18766](https://github.com/opensearch-project/OpenSearch/pull/18766))
+- Fix flaky LeafSorterOptimizationTests by improving segment creation reliability and CI environment compatibility ([#18898](https://github.com/opensearch-project/OpenSearch/issues/18898))
```

Comment:
It's generally fine to [omit a changelog entry](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#which-changes-require-a-changelog-entry) for test-related fixes.